src/platform/plugins/shared/discover/public/context_awareness/profile_providers/register_enabled_profile_providers.test.ts

```diff
@@ -0,0 +1,174 @@
+    jest
+      .spyOn(profileProviderServices.core.pricing, 'isFeatureAvailable')
+      .mockImplementation((featureId) => {
+        if (featureId === FEATURE_ID_1) {
+          return true;
+        }
+        return false;
+      });
+
+    // Sanity check
+    expect(profileProviderServices.core.pricing.isFeatureAvailable(FEATURE_ID_1)).toBeTruthy();
+
+    registerEnabledProfileProviders({
+      profileService: dataSourceProfileServiceMock,
+      providers: [dataSourceProfileProviderMock],
+      enabledExperimentalProfileIds: [],
+      services: profileProviderServices,
+    });
+    const rootContext = await rootProfileServiceMock.resolve({ solutionNavId: null });
+    const dataSourceContext = await dataSourceProfileServiceMock.resolve({
+      rootContext,
+      dataSource: createEsqlDataSource(),
+      query: { esql: 'from my-example-logs' },
+    });
+
+    const profile = dataSourceProfileServiceMock.getProfile({ context: dataSourceContext });
+    const baseImpl = () => ({});
+    profile.getCellRenderers?.(baseImpl)({} as unknown as CellRenderersExtensionParams);
+    expect(dataSourceProfileProviderMock.profile.getCellRenderers).toHaveBeenCalledTimes(1);
+    expect(dataSourceProfileProviderMock.profile.getCellRenderers).toHaveBeenCalledWith(baseImpl, {
+      context: dataSourceContext,
+    });
+  });
+
+  it('should not register restricted profile when product feature is not available', async () => {
+    const {
+      rootProfileServiceMock,
+      dataSourceProfileServiceMock,
+      dataSourceProfileProviderMock,
+      profileProviderServices,
+    } = createContextAwarenessMocks({
+      shouldRegisterProviders: false,
+    });
+
+    // Mock feature availability
+    jest
+      .spyOn(profileProviderServices.core.pricing, 'isFeatureAvailable')
+      .mockImplementation((featureId) => {
+        if (featureId === FEATURE_ID_2) {
+          return true;
+        }
+        return false;
+      });
+
+    // Sanity check
+    expect(profileProviderServices.core.pricing.isFeatureAvailable(FEATURE_ID_1)).not.toBeTruthy();
+
+    registerEnabledProfileProviders({
+      profileService: dataSourceProfileServiceMock,
+      providers: [dataSourceProfileProviderMock],
+      enabledExperimentalProfileIds: [],
+      services: profileProviderServices,
+    });
+    const rootContext = await rootProfileServiceMock.resolve({ solutionNavId: null });
+    const dataSourceContext = await dataSourceProfileServiceMock.resolve({
+      rootContext,
+      dataSource: createEsqlDataSource(),
+      query: { esql: 'from my-example-logs' },
+    });
+
+    const profile = dataSourceProfileServiceMock.getProfile({ context: dataSourceContext });
+    const baseImpl = () => ({});
+    profile.getCellRenderers?.(baseImpl)({} as unknown as CellRenderersExtensionParams);
+    expect(dataSourceProfileProviderMock.profile.getCellRenderers).not.toHaveBeenCalled();
+    expect(profile).toMatchObject({});
+  });
+});
```

Comment:
Extracted as is from `src/platform/plugins/shared/discover/public/context_awareness/profile_providers/register_profile_providers.test.ts`